Source/Core/VideoCommon/VertexLoaderManager.cpp

```diff
@@ -152,6 +155,36 @@ int RunVertices(int vtx_attr_group, int primitive, int count, DataReader src, bo
 	if (skip_drawing || is_preprocess)
 		return size;
 
+	// Object Removal Code code
+	if (m_LocalCoreStartupParameter.num_object_removal_codes)
+	{
+		if (m_LocalCoreStartupParameter.update)
+		{
+			// Set lock so codes can be enabled/disabled in game without crashes.
+			m_LocalCoreStartupParameter.done = false;
+			for (int current_object_removal_code = 0; current_object_removal_code < m_LocalCoreStartupParameter.num_object_removal_codes; current_object_removal_code++)
+			{
+				for (int d = 0; d < m_LocalCoreStartupParameter.num_object_skip_data_bytes[current_object_removal_code]; d++)
+				{
+					if (src.Peek<u8, false>(d) != m_LocalCoreStartupParameter.object_removal_codes[current_object_removal_code][d])
+					{
+						//Data didn't match, try next object_removal_code
+						break;
+					}
+					else
```

Comment:
@degasus If I move the "return size;" after the inner most for loop, everything is skipped, resulting in a black screen.  This is because the first object that doesn't match will break out of that for loop and hit the "return size" line.  It needs to know that it matched every <u8> correctly before "return size;" is executed.

It's true I don't need the "else" on line 174 though, because if it enters the first if statement it will always break out of the for loop.  I could remove the "else" and just have the two if statements.  I might look at the assembly and see if there is any difference between the two, but if the compiler is smart enough, I'd think both would be the same.